### PR TITLE
feat(test-utils): add `mockFn` and `mockLogger` utils

### PR DESCRIPTION
**Pull request by ricardogobbosouza**
### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Add `mockLogger` used in module testing

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/test-utils/src/mocks.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+}
+
+export const mockFnMaps = {
+  jest: mockFnJest,
+  vitest: mockFnVitest
+}
+
+export function mockLogger (): typeof consola {
+  const mock = {}
+  const mockFn = mockFnMaps[useTestContext().options.runner]
```
Comment (pi0) [inline packages/test-utils/src/mocks.ts:21]: Can you please create `mockFn()` utility that does this? By default (when runner is not jest) we can fallback to vitest.

File: packages/test-utils/src/mock.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    return jest.fn()
+  }
+
+  if (runner === 'vitest') {
+    const vitest = await import('vitest')
+    return vitest.vi.fn()
+  }
+
+  return () => {}
+}
```
Comment (danielroe) [inline packages/test-utils/src/mock.ts:20]: Can we move the async import and creation of mock to `createTest` instead so this function can be sync?

**Review by pi0 — APPROVED** [review]
Nice initiative! Thanks (i didn't well tested this addition with latest changes. please feel free to make pr if have any issues)

**Comment by ricardogobbosouza** [comment]
Nice @pi0

**Pull request merged**